### DISCUSSION TITLE
Performance Problems: Map Unresponsive/Slow when panning or zooming & 200-300 markers

### DISCUSSION THREAD

**Discussion by matthiasprieth** [giorgiabosello/google-maps-react-markers#316] [Help] [answered]
Hi, I tried to migrate from _google-map-react/google-map-react_ to this library because of the jumping markers problem in the original library: https://github.com/google-map-react/google-map-react/issues/1143

I was really happy first, but then the performance was a deal breaker when panning/dragging & zooming.
First of all I know I can use clustering with this library, but right now we try not to use it.
The strange thing is that I'm seeing such a better performance with the original library. Do you have any idea why that happens? As soon as we get rid of the markers, performance is no deal anymore. As marker components we just use simple divs without any images or something special. 

So right now we'd rather stay with the jumping then the performance problems. I also want to say that on Firefox the performance problem is less of a deal (on chrome currently). Let me know If you've experienced something like that or if I should create a bug report.

**Comment by giorgiabosello** [comment]
Hi @matthiasprieth, I use the same code of this repository in our clients website. Someone has over 3k stores to show and performance is not a problem. We do use clustering though (and we used them also in _google-map-react/google-map-react_).
Soon I got time, I'll try to create a live version with more stores to see if I can find anything, or PR are welcome if you want to try.

**Reply by matthiasprieth** [reply]
Hi @giorgiabosello, thx for the quick reply. I realized performance drops seem to become noticeable with 100+ markers. Also tried to use simple div's as markers without any styling libraries like @emotion or @materialUI. Also no rerenders happening or something. HTML style changes in the dev tools also seem to behave similar as with the google-map-react/google-map-react module.

Would be awesome if you could quickly test this out on your end in Chrome as well with 100 - 1000 stores... 🙌

**Reply by matthiasprieth** [reply]
Hi @giorgiabosello, have you had a chance to take a look on your end with a lot of stores?

**Reply by giorgiabosello** [reply]
@matthiasprieth not yet, sorry. It has been a very hard period. Winter holidays are coming, I'll take a look in a few weeks. 
Thank you for your patience 🙏🏼

**Reply by giorgiabosello** [reply]
Hi @matthiasprieth I'm looking into it and I can confirm the issue about performances. Hope to find something soon.
[Issue opened](https://github.com/giorgiabosello/google-maps-react-markers/issues/332).

**Reply by giorgiabosello** [reply, marked as answer]
@matthiasprieth try with the 🔖release [2.0.12](https://github.com/giorgiabosello/google-maps-react-markers/releases/tag/2.0.12).
Live example in the repository [homepage](https://giorgiabosello.github.io/google-maps-react-markers/).
